app/utilities/decimal_places.py

```diff
@@ -0,0 +1,124 @@
+from decimal import Decimal
+from typing import Literal, TypeAlias
+
+import flask_babel
+from babel import Locale, numbers, units
+from babel.numbers import get_currency_precision
+
+UnitLengthType: TypeAlias = Literal["short", "long", "narrow"]
+
+
+def custom_format_decimal(value: int | Decimal | float, locale: Locale | str) -> str:
+    """
+    This function provides a wrapper for the numbers `format_decimal` method, generating the
+    number format (including the desired number of decimals), based on the value entered by the user and
+    the locale.
+    """
+    number_format = get_number_format(value, locale)
+
+    return numbers.format_decimal(
+        value,
+        format=number_format,
+        locale=locale,
+    )
+
+
+def get_formatted_currency(
+    *,
+    value: float | Decimal,
+    currency: str = "GBP",
+    locale: str | None = None,
+    decimal_limit: int | None = 0,
```

Comment:
Default should be None
```suggestion
    decimal_limit: int | None = None,
```